depthai_ros_driver/src/depthai_loader.cpp

```diff
@@ -8,23 +8,43 @@
 #include <pluginlib/class_loader.h>
 #include <ros/ros.h>
 
-
 int main(int argc, char** argv) {
-    ros::init(argc, argv, "temp");
+    ros::init(argc, argv, "depthai");
 
     ros::NodeHandle private_nh("~");
 
     std::string pipeline_name = "depthai_ros_driver/StereoWithMobilenetSSDPipeline";
     if (!private_nh.getParam("pipeline", pipeline_name)) {
         private_nh.setParam("pipeline", pipeline_name);
     }
+    std::string device_id;
+    dai::DeviceInfo device_info;
+    if (!private_nh.getParam("device_id", device_id)) {
+        ROS_INFO("Trying to find a device");
+        auto info = rr::DeviceROS::getFirstAvailableDevice();
+        device_info = std::get<1>(info);
+        device_id = device_info.getMxId();
+        if (!std::get<0>(info)) {
+            ROS_WARN("Device not available (mxid = %s)", device_id.c_str());
+        }
+        private_nh.setParam("device_id", device_id);
+        ROS_INFO("Found device: %s", device_id.c_str());
+    } else {
+        auto info = rr::DeviceROS::getDeviceByMxId(device_id);
```

Comment:
replace device_id to mx_id for better clarity?
Also replace the `private_nh.getParam("device_id", device_id)`